### PR TITLE
TagBox: Fix horizontal scrolling with RTL in Chrome 85+

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

**Review by alexander-kotov-dx — DISMISSED** [review]

File: js/ui/tag_box.js [hunk 1]
```diff
@@ -239,8 +239,12 @@ const TagBox = SelectBox.inherit({
     _getBorderPosition: function(direction) {
         const rtlEnabled = this.option('rtlEnabled');
         const isScrollLeft = (direction === 'end') ^ rtlEnabled;
-        const isScrollReverted = rtlEnabled && !browser.webkit;
-        const scrollSign = (!rtlEnabled || browser.webkit || browser.msie) ? 1 : -1;
+
+        // NOTE: for affecting this Chrome BC: https://www.chromestatus.com/feature/5759578031521792
+        const isNewChrome = browser.chrome && +browser.version >= 85;
+        const isOldChrome = browser.chrome && +browser.version < 85;
```
Comment (San4es) [inline js/ui/tag_box.js]:
```suggestion
        const browserVersion = parseInt(browser.version, 10);
        const isChrome =  browser.chrome;
        const isNewChrome = isChrome && browserVersion >= 85;
        const isOldChrome = isChrome && browserVersion < 85;
```

File: js/ui/tag_box.js [hunk 2]
```diff
@@ -239,8 +239,12 @@ const TagBox = SelectBox.inherit({
...
+        const isScrollReverted = rtlEnabled && (!browser.webkit || isNewChrome);
+        const scrollSign = (!rtlEnabled || browser.webkit && !browser.chrome || isOldChrome || browser.msie) ? 1 : -1;
```
Comment (San4es) [inline js/ui/tag_box.js]:
```suggestion
        const scrollSign = (!rtlEnabled || browser.webkit && !isNewChrome || browser.msie) ? 1 : -1;
```
?

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 3]
```diff
@@ -53,6 +53,8 @@ const KEY_SPACE = ' ';
 const CLEAR_BUTTON_AREA = 'dx-clear-button-area';
 
 const TIME_TO_WAIT = 500;
+const isBrowserOldChrome = browser.chrome && +browser.version < 85;
+const isBrowserNewChrome = browser.chrome && +browser.version >= 85;
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
const browserVersion = parseInt(browser.version, 10);
const isBrowserOldChrome = browser.chrome && browserVersion < 85;
const isBrowserNewChrome = browser.chrome && browserVersion >= 85;
```

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 4]
```diff
@@ -5047,9 +5049,9 @@ QUnit.module('single line mode', {
         this.instance.option('rtlEnabled', true);
 
         const $container = this.$element.find('.' + TAGBOX_TAG_CONTAINER_CLASS);
-        const sign = browser.webkit || browser.msie ? 1 : -1;
+        const sign = browser.webkit && !browser.chrome || isBrowserOldChrome || browser.msie ? 1 : -1;
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        const sign = browser.webkit && !isBrowserNewChrome || browser.msie ? 1 : -1;
```
?

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 5]
```diff
@@ -5058,21 +5060,21 @@ QUnit.module('single line mode', {
...
-        assert.equal($container.scrollLeft(), expectedScrollPosition, 'scroll position is correct on focus out');
+        assert.roughEqual($container.scrollLeft(), expectedScrollPosition, 1, 'scroll position is correct on focus out');
     });
 
     QUnit.test('tags container should be scrolled to the end on focusin in the RTL mode (T390041)', function(assert) {
         this.instance.option('rtlEnabled', true);
 
         const $container = this.$element.find('.' + TAGBOX_TAG_CONTAINER_CLASS);
-        const sign = browser.webkit || browser.msie ? 1 : -1;
+        const sign = browser.webkit && !browser.chrome || isBrowserOldChrome || browser.msie ? 1 : -1;
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        const sign = browser.webkit && !isBrowserNewChrome || browser.msie ? 1 : -1;
```

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 6]
```diff
@@ -5218,18 +5220,18 @@ QUnit.module('keyboard navigation through tags in single line mode', {
         });
 
         const $container = this.$element.find('.' + TAGBOX_TAG_CONTAINER_CLASS);
-        const isScrollReverted = (browser.msie || browser.mozilla);
-        const sign = browser.webkit || browser.msie ? 1 : -1;
+        const isScrollReverted = (browser.msie || browser.mozilla || isBrowserNewChrome);
+        const sign = browser.webkit && !browser.chrome || isBrowserOldChrome || browser.msie ? 1 : -1;
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        const sign = browser.webkit && !isBrowserNewChrome || browser.msie ? 1 : -1;
```
?

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 7]
```diff
@@ -5320,11 +5322,11 @@ QUnit.module('keyboard navigation through tags in single line mode', {
             .press('left');
 
         const $container = this.$element.find('.' + TAGBOX_TAG_CONTAINER_CLASS);
-        const isScrollReverted = browser.msie || browser.mozilla;
-        const sign = browser.webkit || browser.msie ? 1 : -1;
+        const isScrollReverted = browser.msie || browser.mozilla || isBrowserNewChrome;
+        const sign = browser.webkit && !browser.chrome || isBrowserOldChrome || browser.msie ? 1 : -1;
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
        const sign = browser.webkit && !isBrowserNewChrome  || browser.msie ? 1 : -1;
```
?

File: js/ui/tag_box.js (hunk 1 above)
Comment (ksercs) [reply js/ui/tag_box.js]: fixed

File: js/ui/tag_box.js (hunk 2 above)
Comment (ksercs) [reply js/ui/tag_box.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 3 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 4 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 5 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 6 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 7 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: fixed

**Review by alexander-kotov-dx — APPROVED** [review]

**Review by San4es — APPROVED** [review]

**Pull request merged**